src/main/java/Storage.java

```diff
@@ -0,0 +1,105 @@
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Represents a storage for users' tasklist.
+ */
+public class Storage {
+    private static String filePath;
+    private ArrayList<Task> taskList;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.taskList = new ArrayList<>();
+    }
+
+    /**
+     * Returns user's tasklist.
+     *
+     * @return arraylist of tasks
+     * @throws IOException occurs when file does not exist or is corrupted
+     */
+    public ArrayList<Task> load() throws IOException {
+        BufferedReader reader = new BufferedReader(new FileReader(filePath));
+        String line = null;
+
+        while ((line = reader.readLine()) != null) {
+            char taskType = line.charAt(1);
+            char isDone = line.charAt(4);
+            if (taskType == 'T') {
+                String description = line.substring(7);
+                Task todo = new ToDo(description);
+                if (isDone == '0') {
+                    taskList.add(todo);
+                } else {
+                    todo.markAsDone();
+                    taskList.add(todo);
+                }
+            } else if (taskType == 'D') {
+                String date = "";
+                String time = "";
+                String description = line.substring(7, line.indexOf('(') - 1);
+                String by = line.substring(line.indexOf(':') + 2, line.indexOf(')')).trim();
+                String[] dateTimeSplit = by.split(" ");
+                if (dateTimeSplit.length == 3) {
+                    date = dateTimeSplit[0] + " " + dateTimeSplit[1] + " "
+                                   + dateTimeSplit[2].substring(0, dateTimeSplit[2].length() - 1);
+                    time = "";
+                } else if (dateTimeSplit.length == 4) {
+                    date = dateTimeSplit[0] + " " + dateTimeSplit[1] + " "
+                                   + dateTimeSplit[2].substring(0, dateTimeSplit[2].length() - 1);
+                    time = dateTimeSplit[3];
+                }
+                Task deadline = new Deadline(description, new DateTime(date, time));
+                if (isDone == '0') {
+                    taskList.add(deadline);
+                } else {
+                    deadline.markAsDone();
+                    taskList.add(deadline);
+                }
+            } else if (taskType == 'E') {
+                String date = "";
+                String time = "";
```

Comment:
Seems like there is a repeat of code for when taskType == "E" and taskType == "D".
Perhaps you could abstract out the code in a function for easier reuse.